### DISCUSSION TITLE
Problem with install req

### DISCUSSION THREAD

**Discussion by sarboys** [xai-org/grok-1#58] [General]
Mackbook air M1. 
Python and pip last version.

pip3 install -r requirements.txt

WARNING: jax 0.4.25 does not provide the extra 'cuda12-pip'
INFO: pip is looking at multiple versions of jax[cuda12-pip] to determine which version is compatible with other requirements. This could take a while.
ERROR: Could not find a version that satisfies the requirement jaxlib==0.4.25+cuda12.cudnn89; extra == "cuda12_pip" (from jax[cuda12-pip]) (from versions: 0.4.3, 0.4.4, 0.4.6, 0.4.7, 0.4.9, 0.4.10, 0.4.11, 0.4.12, 0.4.13, 0.4.14, 0.4.16, 0.4.17, 0.4.18, 0.4.19, 0.4.20, 0.4.21, 0.4.22, 0.4.23, 0.4.24, 0.4.25)
ERROR: No matching distribution found for jaxlib==0.4.25+cuda12.cudnn89; extra == "cuda12_pip"

**Comment by konard** [comment]
There is an issue exactly about that problem: https://github.com/xai-org/grok-1/issues/6

**Reply by NeweggWolf** [reply]
try to change "jax[cuda12_pip]==0.4.25" to "jax[cuda12-pip]==0.4.25"

**Comment by StavrosD** [comment]
This worked for manually installation, I cannot install the requirements.txt. Maybe we have to modify the requirements.txt
`
pip install -U "jax[cuda12_pip]" -f https://storage.googleapis.com/jax-rele
ases/jax_cuda_releases.html
`
Update
I noticed that you use a Mac. Cuda is not available on Mac computers. 
Mac computers use their own CPU and the unified memory for loading the AI models. Even if you make modifications and run it on the MAC, there is no way to load the model on the memory, it requires about 300GB of memory. 

Best case scenario is running the model directly from the SSD, if someone modify the code. It will be much slower in that case.
